[AN] 강제 업데이트 다이얼로그가 가려지는 문제 수정

moondev03 opened the pull request:
<!-- PR 제목 예시: [AN] 로그인 화면 UI 구현 / [BE] 회원가입 API 추가 / [ALL] 공통 유틸 정리 -->

## ✅ 작업 개요
<!-- 어떤 기능/수정/리팩토링인지 한 줄로 설명 (예: 로그인 화면 UI 구현) -->
- Flow 기반으로 변경
- 강제 업데이트가 필요한 경우 Splash 화면을 종료하도록 변경
- 스토어를 플레이 스토어로 타겟팅

<br>

## 🎯 관련 이슈
<!-- 관련된 이슈 번호를 태깅해주세요. -->
- Closed #649 

<br>

## 🔍 상세 내용
<!-- 
작업한 내용을 구체적으로 작성해주세요.
- 로그인 화면 UI 구성 (이메일, 비밀번호 입력란, 로그인 버튼)
- 입력값 유효성 검사 추가
- ViewModel 및 상태 관리 로직 연동
-->
### Flow 기반으로 변경
MainActivity, MainViewModel Flow 기반으로 마이그레이션

### 문제 해결
강제 업데이트 다이얼로그가 Splash 화면에 가려지는 문제가 있었는데,
강제 업데이트가 필요한 경우 Splash 화면을 종료하도록 변경하였습니다.

### 플레이스토어 타겟팅
package를 추가하여 플레이스토어만 타겟팅 되도록 하였습니다.
```kotlin
// 이전
intent.data = "market://details?id=$packageName".toUri()

// 이후
intent.data = "market://details?id=$packageName".toUri()
intent.`package` = "com.android.vending"
```




<br>

<br>

## 💬 기타 참고사항
<!-- 
코드 리뷰어가 참고해야 할 사항이 있다면 적어주세요.
- 추후 `회원가입` 화면에서도 재사용 가능한 UI 컴포넌트로 분리할 예정입니다.
- `AuthViewModel` 내 중복 로직은 이후 리팩토링에서 정리 예정입니다.
-->

<br>


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

## 릴리스 노트

* **개선 사항**
  * UI 이벤트 처리 비동기화 — 앱 상태 수집 방식 개선으로 화면 반응성 향상
  * 로그인·등록 흐름 및 푸시 토큰 처리 개선 — 인증 준비 상태와 재시도 안정성 증가
  * 수명주기 기반 실행 보강 — 리소스 안전성 개선
  * Google Play 스토어 연결 강화 — 스토어로 직접 이동

* **변경 사항**
  * UI 상태 단순화 — 불필요한 로딩 플래그 제거로 상태 관리 간소화

* **신규 기능**
  * 권한 저장 기능 추가 — 사용자 권한 선택을 앱에 저장하여 재사용
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

moondev03 committed:
refactor: MainViewModel 로직 순서 및 구조 개선

- `FlowBaseViewModel`을 상속하도록 변경합니다.
- 함수들의 호출 순서를 `checkForceUpdate` -> `checkPermissionFlag` -> `checkRegisteredMember` 순으로 명확하게 정리합니다.
- `isLoading` 상태를 제거하고, 모든 로직 완료 후 `isReady`를 true로 설정하여 UI가 표시되도록 변경합니다.
- FCM 토큰을 가져오는 로직을 `fetchFcmToken` 함수로 분리하여 재사용성을 높입니다.

moondev03 committed:
refactor: MainUiState에서 불필요한 isLoading 상태 제거

moondev03 committed:
refactor: LiveData를 StateFlow로 마이그레이션

- `MainActivity`에서 `LiveData`를 `StateFlow`로 변경하여 `repeatOnLifecycle`을 사용해 UI 이벤트를 수집하도록 수정했습니다.
- Play Store를 실행할 때 `com.android.vending` 패키지를 명시하여, 여러 앱 스토어가 설치된 경우에도 Google Play Store가 실행되도록 보장합니다.

moondev03 committed:
refactor: 주석 처리 수정

moondev03 committed:
refactor: 사용하지 않는 import 제거

cucumber99 commented on android/Bottari/presentation/src/main/java/com/bottari/presentation/view/main/MainActivity.kt:
```diff
@@ -54,17 +57,21 @@ class MainActivity : BaseActivity<ActivityMainBinding>(ActivityMainBinding::infl
-                MainUiEvent.SavePermissionFlagFailure,
-                -> finishAffinity()
-            }
+        lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) { viewModel.uiEvent.collect(::handleUiEvent) }
+        }
```
C: `BaseActivity`와 `BaseFragment`에 `collectWithLifecycle` 메소드가 있긴 해요

moondev03 committed:
refactor: MainViewModel 로직 흐름 개선 및 코드 간소화

- `checkForceUpdateUseCase` 호출 시 `invokeOnCompletion`을 사용하지 않고, 강제 업데이트가 필요할 때와 아닐 때 모두 로직의 시작 부분에서 `isReady` 상태를 `true`로 설정하도록 변경하여 흐름을 개선했습니다.
- `.onSuccess { ... }` 형태의 람다식을 메소드 참조(`::`)로 변경하여 코드를 간소화했습니다.
